src/main/java/org/gradle/java/JigsawPlugin.java

```diff
@@ -83,214 +82,149 @@ public void execute(Project project) {
-                args.add("--module");
-                args.add(String.format("%s/%s", module.getName(), run.getMain()));
-                run.setJvmArgs(args);
+                args.add(compileJava.getClasspath().getAsPath());
+                args.add("-d");
+                args.add(moduleDir);
+                compileJava.getOptions().setCompilerArgs(args);
+                compileJava.setClasspath(project.files());
+            }
+        });
+        compileJava.doLast(new Action<Task>() {
+            @Override
+            public void execute(Task task) {
+                project.copy(new Action<CopySpec>() {
+                    @Override
+                    public void execute(CopySpec copySpec) {
+                        copySpec.from(moduleDir);
+                        copySpec.into(compileJava.getDestinationDir());
+                    }
+                });
             }
         });
     }
 
-    private void configureStartScriptsTask(Project project) {
-        final CreateStartScripts startScripts = getStartScriptsTask(project);
-        final SourceSet main = ((SourceSetContainer) project.getProperties().get("sourceSets")).getByName("main");
+    private void configureCompileTestJavaTask(final Project project) {
+        final JavaCompile compileTestJava = (JavaCompile) project.getTasks()
+                .findByName(JavaPlugin.COMPILE_TEST_JAVA_TASK_NAME);
         final SourceSet test = ((SourceSetContainer) project.getProperties().get("sourceSets")).getByName("test");
         final JigsawModule module = (JigsawModule) project.getExtensions().getByName(EXTENSION_NAME);
-        startScripts.doFirst(new Action<Task>() {
+        compileTestJava.getInputs().property("moduleName", module.getName());
+        compileTestJava.doFirst(new Action<Task>() {
             @Override
             public void execute(Task task) {
-                startScripts.setClasspath(project.files());
                 List<String> args = new ArrayList<>();
                 args.add("--module-path");
-                args.add(LIBS_PLACEHOLDER);
-                args.add("--module");
-                args.add(String.format("%s/%s", module.getName(), startScripts.getMainClassName()));
-                startScripts.setDefaultJvmOpts(args);
-            }
-        });
-        startScripts.doLast(new Action<Task>() {
-            @Override
-            public void execute(Task task) {
-                File bashScript = new File(startScripts.getOutputDir(), startScripts.getApplicationName());
-                replaceLibsPlaceHolder(bashScript.toPath(), "\\$APP_HOME/lib");
-                File batFile = new File(startScripts.getOutputDir(), startScripts.getApplicationName() + ".bat");
-                replaceLibsPlaceHolder(batFile.toPath(), "%APP_HOME%\\lib");
+                args.add(compileTestJava.getClasspath().getAsPath());
+                args.add("--add-modules");
+                args.add("junit");
```

Comment:
We may be able to support different frameworks by checking `Test#getTestFramework`.